src/RestControllers/FHIR/FhirExportRestController.php

```diff
@@ -0,0 +1,491 @@
+     */
+    private function getExportServiceRegistry()
+    {
+        if (!empty($this->resourceRegistry)) {
+            return $this->resourceRegistry;
+        }
+        $serviceLocator = new FhirExportServiceLocator($this->request->getRestConfig());
+        $this->resourceRegistry = $serviceLocator->findExportServices();
+        return $this->resourceRegistry;
+    }
+
+    /**
+     * Return the fully populated export job for the given client and user.
+     * @param $jobId The unique identifier for the job
+     * @param $clientId The api client the job belongs to
+     * @param $userId The user that created the job request
+     * @return ExportJob
+     * @throws \InvalidArgumentException if the $jobId, $clientId, or $userId is invalid
+     */
+    private function getJobForId($jobId, $clientId, $userId)
+    {
+        // TODO: @adunsulag when we have the client tied to a system user for Credentials grant we will attach it here.
+        $sql = "SELECT `uuid`, `start_time`, `resource_include_time`, `output_format`, `resources`, "
+            . "`client_id`, `user_id`, `access_token_id`, `status`, `request_uri`, `output`, `errors` "
+            . "FROM `export_job` WHERE `uuid` = ? AND `client_id`=? AND `user_id` = ? ";
+
+        $params = [$jobId, $clientId, $userId];
+        $ret = sqlQueryNoLog($sql, $params);
+
+        if (empty($ret)) {
+            $this->logger->error(
+                "FhirExportRestController->getJobForId() failed to find job",
+                ['jobId' => $jobId, 'sql' => $sql, 'params' => $params]
+            );
+            throw new \InvalidArgumentException("Export Job with jobId '" . $jobId . "' does not exist");
+        }
+
+        $this->logger->debug("FhirExportRestController->getJobForId() ", ['jobId' => $jobId, 'dbResult' => $ret]);
+
+        $job = new ExportJob();
+        $job->setId($jobId);
+        $job->setStartTime(\DateTime::createFromFormat(self::DATETIME_FORMAT, $ret['start_time']));
+        $job->setResourceIncludeTime(\DateTime::createFromFormat(self::DATETIME_FORMAT, $ret['resource_include_time']));
+        $job->setOutputFormat($ret['output_format']);
+        $job->setResources($ret['resources']);
+        $job->setClientId($ret['client_id']);
+        $job->setUserId($ret['user_id']);
+        $job->setAccessTokenId($ret['access_token_id']);
+        $job->setStatus($ret['status']);
+        $job->setRequestURI($ret['request_uri']);
+        $job->setOutput($ret['output']);
+        $job->setErrors($ret['errors']);
+        return $job;
+    }
+
+    /**
+     * Given an export job, save it to the database
+     * @param ExportJob $job The job to save
+     * @return ExportJob the saved job
+     * @throws \RuntimeException if the job fails to save
+     */
+    private function createJobRequest(ExportJob $job)
+    {
+        // we will generate a UUID here, if we ever want the db to do that we would accomplish that here...
```

Comment:
would use OpenEMR's custom uuid maker, which will create a hybrid ordered/random uuid. Involves:
```
use OpenEMR\Common\Uuid\UuidRegistry;
```
then:
```
$binaryUuid = (new UuidRegistry(['table_name' => 'export_job']))->createUuid();
```

Reply:
Switched to this format.